### PR TITLE
Add a release pipeline ⛏

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This will need to be run manually for the first(s) releases, but it
sets pieces in place and help the dogfooding effort for Tekton
Pipelines.

It reuses golang tasks from the tektoncd/catalog *and* adds a
`goreleaser` Task (using goreleaser).

The pipeline runs the lint, then build the command and run the unit
tests. It finaly cross build the binary for all targeted architecture
and system and publish a draft release.

Although this should be considered still a work in progress, it is
still usable and will be run to make the v0.1.0 release of
tektoncd/cli.

/cc @sthaha @chmouel 

This is *slightly* blocked by #77 (although both can be merged asynchronously)

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ :no_good_man: ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/release-pipeline-run.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+kind: PipelineResource
+metadata:
+  name: tektoncd-cli-git
+spec:
+  type: git
+  params:
+  - name: revision
+    value: master
+  - name: url
+    value: https://github.com/vdemeester/tektoncd-cli
```
Comment (chmouel) [inline tekton/release-pipeline-run.yml]: should be the official path?

**Comment by vdemeester** [comment]
@sthaha @chmouel @hrishin this can be test if : 
1. you modify the `PipelineResource` to target your fork
2. you create and push a *dummy* tag on your fork
3. you create a secret based on a token that has `repo:admin` as security clearance :angel:

File: tekton/release-pipeline-run.yml (hunk 1 above)
Comment (vdemeester) [reply tekton/release-pipeline-run.yml]: ahaha yes, I though I've updated this one before pushing :man_facepalming:

Comment (vdemeester) [reply tekton/release-pipeline-run.yml]: Fixed :angel:

**Pull request merged**